packer crash when using a json variables file

I'm not sure if its a configuration issue, but this is the crash that is occurring when I try to test changes from my vsphere_export_ovf branch. I built it using `go install`

```
$ packer build -on-error=ask -var-file variables.json ubuntu-18.04-amd64-docker.json
2020/03/19 00:06:27 [INFO] Packer version: 1.5.5-dev [go1.14 windows amd64]
2020/03/19 00:06:27 Checking 'PACKER_CONFIG' for a config file path
2020/03/19 00:06:27 'PACKER_CONFIG' not set; checking the default config file path
2020/03/19 00:06:27 Attempting to open config file: C:\Users\jhawk28\AppData\Roaming\packer.config
2020/03/19 00:06:27 [WARN] Config file doesn't exist: C:\Users\jhawk28\AppData\Roaming\packer.config
2020/03/19 00:06:27 Setting cache directory: C:\Users\jhawk28\repos\infrastructure\packer\packer_cache
cannot determine if process is in background: Process background check error: not implemented yet
2020/03/19 00:06:27 waiting for all plugin processes to complete...
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/hashicorp/packer/command.(*Meta).Core(0xc00007c460, 0xc0004e20b0, 0xc0004e20b0, 0x0, 0x0)
        C:/Users/jhawk28/repos/packer/command/meta.go:54 +0x16a
github.com/hashicorp/packer/command.(*BuildCommand).GetBuilds(0xc00007c460, 0xc0000466c0, 0x1e, 0x6, 0x1, 0x7fffffffffffffff, 0xc00004e11a)
        C:/Users/jhawk28/repos/packer/command/build.go:155 +0x196
github.com/hashicorp/packer/command.(*BuildCommand).RunContext(0xc00007c460, 0x47a5700, 0xc000078200, 0xc000138020, 0x4, 0x6, 0xc000078200)
        C:/Users/jhawk28/repos/packer/command/build.go:185 +0xd1
github.com/hashicorp/packer/command.(*BuildCommand).Run(0xc00007c460, 0xc000138020, 0x4, 0x6, 0x0)
        C:/Users/jhawk28/repos/packer/command/build.go:55 +0x1d6
github.com/mitchellh/cli.(*CLI).Run(0xc0000cc3c0, 0xc0000cc3c0, 0x1, 0x1)
        C:/Users/jhawk28/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1e1
main.wrappedMain(0x0)
        C:/Users/jhawk28/repos/packer/main.go:238 +0x828
main.realMain(0x0)
        C:/Users/jhawk28/repos/packer/main.go:124 +0xec
main.main()
        C:/Users/jhawk28/repos/packer/main.go:34 +0x29
```

--- a/command/build_test.go
+++ b/command/build_test.go
@@ -53,13 +53,23 @@ func TestBuild_VarArgs(t *testing.T) {
 		},
 
 		{
-			name: "hcl - inexistent var file errs",
+			name: "hcl - inexistent json var file errs",
 			args: []string{
 				"-var-file=" + filepath.Join(testFixture("var-arg"), "potato.json"),
 				testFixture("var-arg"),
 			},
 			expectedCode: 1,
 			fileCheck:    fileCheck{notExpected: []string{"potato.txt"}},
+		},
+
+		{
+			name: "hcl - inexistent hcl var file errs",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "potato.hcl"),
+				testFixture("var-arg"),
+			},
+			expectedCode: 1,
+			fileCheck:    fileCheck{notExpected: []string{"potato.hcl"}},
 		},
 
 		{

--- a/command/build_test.go
+++ b/command/build_test.go
@@ -17,6 +17,96 @@ import (
 	shell_local "github.com/hashicorp/packer/provisioner/shell-local"
 )
 
+func TestBuild_VarArgs(t *testing.T) {
+	tc := []struct {
+		name         string
+		args         []string
+		expectedCode int
+		fileCheck
+	}{
+		{
+			name: "json - json varfile sets an apple env var",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "apple.json"),
+				filepath.Join(testFixture("var-arg"), "fruit_builder.json"),
+			},
+			fileCheck: fileCheck{expected: []string{"apple.txt"}},
+		},
+
+		{
+			name: "json - arg sets a pear env var",
+			args: []string{
+				"-var=fruit=pear",
+				filepath.Join(testFixture("var-arg"), "fruit_builder.json"),
+			},
+			fileCheck: fileCheck{expected: []string{"pear.txt"}},
+		},
+
+		{
+			name: "json - inexistent var file errs",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "potato.json"),
+				filepath.Join(testFixture("var-arg"), "fruit_builder.json"),
+			},
+			expectedCode: 1,
+			fileCheck:    fileCheck{notExpected: []string{"potato.txt"}},
+		},
+
+		{
+			name: "hcl - inexistent var file errs",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "potato.json"),
+				testFixture("var-arg"),
+			},
+			expectedCode: 1,
+			fileCheck:    fileCheck{notExpected: []string{"potato.txt"}},
+		},
+
+		{
+			name: "hcl - auto varfile sets a chocolate env var",
+			args: []string{
+				testFixture("var-arg"),
+			},
+			fileCheck: fileCheck{expected: []string{"chocolate.txt"}},
+		},
+
+		{
+			name: "hcl - hcl varfile sets a apple env var",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "apple.hcl"),
+				testFixture("var-arg"),
+			},
+			fileCheck: fileCheck{expected: []string{"apple.txt"}},
+		},
+
+		{
+			name: "hcl - json varfile sets a apple env var",
+			args: []string{
+				"-var-file=" + filepath.Join(testFixture("var-arg"), "apple.json"),
+				testFixture("var-arg"),
+			},
+			fileCheck: fileCheck{expected: []string{"apple.txt"}},
+		},
+
+		{
+			name: "hcl - arg sets a tomato env var",
+			args: []string{
+				"-var=fruit=tomato",
+				testFixture("var-arg"),
+			},
+			fileCheck: fileCheck{expected: []string{"tomato.txt"}},
+		},
+	}
+
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			run(t, tt.args, tt.expectedCode)
+			defer cleanup()
+			tt.fileCheck.verify(t)
+		})
+	}
+}
+
 func TestBuildOnlyFileCommaFlags(t *testing.T) {
 	c := &BuildCommand{
 		Meta: testMetaFile(t),
@@ -214,6 +304,35 @@ func TestBuildWithNonExistingBuilder(t *testing.T) {
 	}
 	if fileExists("vanilla.txt") {
 		t.Errorf("NOT expected to find vanilla.tx")
+	}
+}
+
+func run(t *testing.T, args []string, expectedCode int) {
+	t.Helper()
+
+	c := &BuildCommand{
+		Meta: testMetaFile(t),
+	}
+
+	if code := c.Run(args); code != expectedCode {
+		fatalCommand(t, c.Meta)
+	}
+}
+
+type fileCheck struct {
+	expected, notExpected []string
+}
+
+func (fc fileCheck) verify(t *testing.T) {
+	for _, f := range fc.expected {
+		if !fileExists(f) {
+			t.Errorf("Expected to find %s", f)
+		}
+	}
+	for _, f := range fc.notExpected {
+		if fileExists(f) {
+			t.Errorf("Expected to not find %s", f)
+		}
 	}
 }
 

--- a/command/meta.go
+++ b/command/meta.go
@@ -50,6 +50,9 @@ func (m *Meta) Core(tpl *template.Template) (*packer.Core, error) {
 			return nil, err
 		}
 	}
+	if m.flagVars == nil {
+		m.flagVars = map[string]string{}
+	}
 	for k, v := range *fj {
 		m.flagVars[k] = v
 	}

--- a/command/test-fixtures/var-arg/apple.hcl
+++ b/command/test-fixtures/var-arg/apple.hcl
@@ -1,0 +1,2 @@
+
+fruit = "apple"

--- a/command/test-fixtures/var-arg/apple.json
+++ b/command/test-fixtures/var-arg/apple.json
@@ -1,0 +1,3 @@
+{
+    "fruit": "apple"
+}

--- a/command/test-fixtures/var-arg/chocolate.auto.pkrvars.hcl
+++ b/command/test-fixtures/var-arg/chocolate.auto.pkrvars.hcl
@@ -1,0 +1,2 @@
+
+fruit = "chocolate" // is that even a fruit !?

--- a/command/test-fixtures/var-arg/fruit_builder.json
+++ b/command/test-fixtures/var-arg/fruit_builder.json
@@ -1,0 +1,20 @@
+{
+    "variables": {
+        "fruit": ""
+    },
+    "builders": [
+        {
+            "communicator": "none",
+            "type": "null"
+        }
+    ],
+    "post-processors": [
+        [
+            {
+                "name": "apple",
+                "type": "shell-local",
+                "inline": [ "echo {{ user `fruit` }} > {{ user `fruit` }}.txt" ]
+            }
+        ]
+    ]
+}

--- a/command/test-fixtures/var-arg/fuit_builder.pkr.hcl
+++ b/command/test-fixtures/var-arg/fuit_builder.pkr.hcl
@@ -1,0 +1,18 @@
+
+variable "fruit" {
+    type = string
+}
+
+source "null" "builder" {
+    communicator = "none"
+}
+
+build {
+    sources = [ 
+        "source.null.builder",
+    ]
+
+    provisioner "shell-local" {
+        inline = ["echo ${var.fruit} > ${var.fruit}.txt"]
+    }
+}

--- a/hcl2template/parser.go
+++ b/hcl2template/parser.go
@@ -130,11 +130,17 @@ func (p *Parser) parse(filename string, varFiles []string, argVars map[string]st
 		for _, filename := range hclVarFiles {
 			f, moreDiags := p.ParseHCLFile(filename)
 			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
 			varFiles = append(varFiles, f)
 		}
 		for _, filename := range jsonVarFiles {
 			f, moreDiags := p.ParseJSONFile(filename)
 			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
 			varFiles = append(varFiles, f)
 		}
 

